apps/ai-content-generator/src/hooks/config/useInitializeParameters.ts

```diff
@@ -0,0 +1,28 @@
+import { Dispatch } from 'react';
+import { useQuery } from 'react-query';
+import { useSDK } from '@contentful/react-apps-toolkit';
+import type { ConfigAppSDK } from '@contentful/app-sdk';
+import getCurrentParameters from '@react-query/queries/getCurrentParameters';
+import { AppInstallationParameters } from '@/components/config/ConfigForm';
+import { ParameterAction, ParameterActionType } from '@components/config/parameterReducer';
+
+const useInitializeParameters = (dispatch: Dispatch<ParameterActionType>) => {
+  const sdk = useSDK<ConfigAppSDK>();
+
+  const dispatchParameters = (parameters: AppInstallationParameters) => {
+    if (!parameters) return;
+
+    dispatch({
+      type: ParameterAction.CONTENTFUL_PARAMETERS,
+      value: parameters,
+    });
+
+    sdk.app.setReady();
+  };
+
+  useQuery(['appParameters'], getCurrentParameters(sdk), {
```

Comment:
does this value need to be returned from the hook? Not all the way familiar with React Query

Reply:
All good! So when the query resolves (the query in this case is getCurrentParameters) it'll call `dispatchParameters` which will update the state of the app installation parameters the parent.

This is the code from the next PR with the hook being used:

```
const ConfigForm = () => {
  const [parameters, dispatchParameters] = useReducer(parameterReducer, initialParameters);

  useSaveConfigHandler(parameters);
  useInitializeParameters(dispatchParameters);
  
...
```

So the OOO is:
1. Component Loads
2. `useInitializeParameters` runs
3. `getCurrentParemeters` resolves with the InstallationParameters
4. The InstallationParameters are dispatched
5. `parameters` is now updated with the current InstallationParameters from Contentful

Reply:
makes sense!